### PR TITLE
Add documentation around valid package names

### PR DESCRIPTION
**Pull request by dlannoye**
This updating the documentation so other don't run into the same issues I did with #1286

### CODE REVIEW

File: docs/development/configuration.md [hunk 1]
```diff
@@ -204,7 +204,7 @@ note: TypeScript 2.4+ is required for custom transformers.
 Your default package name is `default`. You don't need to change it if you are not planning on having isolated bundles.
 Any bundle added as a script tag will share the `default` package, keep that in mind. If you want to release a package (say to NPM), you probably would want set a different name (to avoid scope collision)
 
-It's imperative to have a __unique name__ (matching an npm package) when publishing a bundle to NPM.
+It's imperative to have a __unique name__ (matching an npm package) when publishing a bundle to NPM. Your package name must start with either an '@' or a lowercase a-z character.
```
Comment (nchanged) [inline docs/development/configuration.md]: I don't think it can start with `@` since it's a symbol used to point to a scoped repository.

Comment (dlannoye) [reply docs/development/configuration.md]: Take look at the [code in the LoaderAPI]( https://github.com/fuse-box/fuse-box/blob/9e0eb11d46f2c5c5d41299ff718af32d7c663d29/src/loader/LoaderAPI.ts#L127). This is the exact check it does:
```
    // https://www.cambiaresearch.com/articles/15/javascript-char-codes-key-codes
    // basically lowcase alphabet starts with 97 ends with 122, and symbol @ is 64
    // which 2x faster than /^([@a-z].*)$/
    if (n >= 97 && n <= 122 || n === 64) {
```

Comment (dlannoye) [reply docs/development/configuration.md]: I work with scoped packages and when I name them something like `@foo/bar` the generated folder structure in node_modules is something like:
```
`@foo`
   -> `bar`
```
And you can definitely import from them by doing `import "@foo/bar"` there is no need for an extra '@'.

Am I misunderstanding your concern? 

Also I can confirm fuse-box works fine with package names beginning with a scope defined in the config. Take a look at this sample:
https://github.com/dlannoye/fuse-box-default-package/tree/scoped-package

Comment (nchanged) [reply docs/development/configuration.md]: Right, I would recommend you to avoid this package option and use Quantum instead tho ;-)

**Pull request closed without merging**